features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Blacklist/List.jsx

```diff
@@ -95,6 +98,77 @@ export default function ListBlacklistThrottlingPolicies() {
         }),
     };
 
+    /**
+ * Mock API call
+ * @returns {Promise}.
+ */
+    function apiCall() {
+        let policyList;
+        let incrementId = 0;
+        return new Promise(((resolve, reject) => {
+            restApi.blacklistPoliciesGet().then((result) => {
+                policyList = result.body.list;
+                const blacklistPolicies = policyList.map((obj) => {
+                    let array = [];
+                    incrementId++;
+                    if (obj.conditionValue === Object(obj.conditionValue)) {
+                        Object.keys(obj.conditionValue);
+                        Object.values(obj.conditionValue);
+                        array = Object.entries(obj.conditionValue);
+                    } else {
+                        array.push(obj.conditionValue);
+                    }
+                    return {
+                        conditionId: incrementId,
+                        conditionUUID: obj.conditionId,
+                        conditionType: obj.conditionType,
+                        conditionValue: array,
+                        conditionStatus: obj.conditionStatus,
+                    };
+                });
+                setBlacklistPolicyList(blacklistPolicies);
+                resolve(blacklistPolicies);
+            }).catch((error) => {
+                reject(error);
+            });
+        }));
+    }
+
+    const handleConditionStatus = (conditionUUID, event) => {
+        let editedConditionId;
+        let editedConditionStatus;
+        const blacklistPolicyListNew = cloneDeep(blacklistPolicyList);
+        blacklistPolicyListNew.map((res) => {
+            if (res.conditionUUID === conditionUUID) {
+                res.conditionStatus = event.target.checked;
+                editedConditionId = conditionUUID;
+                editedConditionStatus = event.target.checked;
+            }
+            return res.conditionStatus;
```

Comment:
If you only return the `res.conditionStatus` what is the purpose of setting other `res` properties in above `if` condition ?

Reply:
editedConditionId and editedConditionStatus are needed to identify the conditionID and Condition Status of the edited Policy. These two values are passed to the API call.

Reply:
fixed with commit `ddbed19`